trick_source/sim_services/MemoryManager/MemoryManager_restore.cpp

```diff
@@ -22,7 +22,7 @@ int Trick::MemoryManager::read_checkpoint( std::istream *is) {
     }
 
     // Search for stls and restore them
-    for ( pos=alloc_info_map.begin() ; pos!=alloc_info_map.end() ; pos++ ) {
+    for ( pos=alloc_info_map.begin() ; do_restore_stls && pos!=alloc_info_map.end() ; pos++ ) {
```

Comment:
Move the test for do_restore_stls to the outside of the loop so it is only checked once.